lib/util/entrypoints.js

```diff
@@ -549,9 +566,10 @@ function walkPath(root, path, target) {
 		node.folder = target;
 	} else {
 		const file = lastNonSlashIndex > 0 ? path.slice(lastNonSlashIndex) : path;
-		if (file.endsWith("*")) {
+		const wildcardsIndex = file.indexOf("*");
```

Comment:
Weak check, in some filesystem file can have `*` character in the filename, let's use more strick checks, i.e. `*` or `*.js`